fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    cplprint_flag = (trim(value)=="true")
+    write(msgString,'(A,l6)') trim(subname)//' cplprint_flag = ',cplprint_flag
+    call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO, rc=rc)
+
+    ! Read in cap debug flag
+    call NUOPC_CompAttributeGet(gcomp, name='dbug_flag', value=value, isPresent=isPresent, isSet=isSet, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+    if (isPresent .and. isSet) then
+     read(value,*) dbug
+    end if
+    write(msgString,'(A,i6)') trim(subname)//' dbug = ',dbug
+    call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO, rc=rc)
+
+  end subroutine
+
+  !-----------------------------------------------------------------------------
+
+  subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+
+    type(ESMF_GridComp)                    :: gcomp
+    type(ESMF_State)                       :: importState, exportState
+    type(ESMF_Clock)                       :: clock
+    integer, intent(out)                   :: rc
+
+!
+! local variables
+    type(ESMF_VM)                          :: vm, fcstVM
+    type(ESMF_Time)                        :: CurrTime, starttime, StopTime
+    type(ESMF_Time)                        :: alarm_output_hf_ring, alarm_output_ring
+    type(ESMF_Time)                        :: alarm_output_hf_stop, alarm_output_stop
+    type(ESMF_TimeInterval)                :: RunDuration, timeStep, rsthour, IAU_offsetTI
+    type(ESMF_Config)                      :: cf
+    type(ESMF_RegridMethod_Flag)           :: regridmethod
+    type(ESMF_TimeInterval)                :: earthStep
+    integer(ESMF_KIND_I4)                  :: nhf, nrg
+
+    integer,dimension(6)                   :: date, date_init
+    integer                                :: mpi_comm_atm
+    integer                                :: i, j, k, io_unit, urc, ierr
+    integer                                :: petcount, mype
+    integer                                :: num_output_file
+    logical                                :: OPENED
+    character(ESMF_MAXSTR)                 :: name
+    logical                                :: fcstpe
+    logical,dimension(:), allocatable      :: wrtpe
+    integer,dimension(:), allocatable      :: petList, originPetList, targetPetList
+    character(20)                          :: cwrtcomp
+    character(160)                         :: msg
+    integer                                :: isrcTermProcessing
+
+    character(len=*),parameter  :: subname='(fv3_cap:InitializeAdvertise)'
+    integer nfmout, nfsout , nfmout_hf, nfsout_hf
+    real(kind=8) :: MPI_Wtime, timewri, timeis,timeie,timerhs, timerhe
+!
+!------------------------------------------------------------------------
+!
+    rc = ESMF_SUCCESS
+    timeis = mpi_wtime()
```

Comment:
Consider `ESMF_TraceRegionEnter()` and `ESMF_TraceRegionExit()` for collecting timing profiles in a single file.